ils_middleware/dags/ingest/record.py

```diff
@@ -0,0 +1,71 @@
+"""Record Loader Workflow for a single file."""
+import logging
+
+from datetime import datetime
+
+from airflow.decorators import dag, task
+from airflow.operators.python import get_current_context
+
+from ils_middleware.tasks.amazon.bluecore_records_s3 import get_file
+from ils_middleware.tasks.bluecore.batch import is_zip, parse_file_to_graph
+from ils_middleware.tasks.bluecore.storage import get_bluecore_db, store_bluecore_resources
+
+logger = logging.getLogger(__name__)
+
+@dag(schedule_interval=None, 
+     start_date=datetime(2025, 2, 20), 
+     catchup=False,
+     tags=['ingest', 'record'],
+     default_args={"owner": "airflow"},)
+def resource_loader():
+    @task()
+    def ingest():
+        context = get_current_context()
+        params = context.get("params")
+        if params is None:
+            raise ValueError("Missing params")
+        file_path = params.get("file")
+        if file_path is None:
+            raise ValueError("Missing file path")
+        logger.info(f"Ingesting {file_path}")
+        local_file_path = get_file(file=file_path)
+        return local_file_path
+
+    @task.branch(task_id="process_choice")
+    def choose_processing(**kwargs):
+        file_path = kwargs.get("file")
+        if is_zip(file_path):
+            return ['process_zip']
+        return ['process']
+
```

Comment:
Should we be defining return types for these functions? I guess it depends on whether we expect to use mypy?

Reply:
Yep, thanks for pointing this out! I think we should define return types. Done in a follow-up commit.